server/jetstream_api.go

```diff
@@ -247,244 +349,282 @@ func (s *Server) jsCreateTemplateRequest(sub *subscription, c *client, subject,
+		return
 	}
-	s.sendAPIResponse(c, subject, reply, string(msg), response)
+
+	resp.StreamInfo = &StreamInfo{State: mset.State(), Config: mset.Config()}
+	s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(resp))
 }
 
 // Request for the list of all streams.
 func (s *Server) jsStreamListRequest(sub *subscription, c *client, subject, reply string, msg []byte) {
 	if c == nil || c.acc == nil {
 		return
 	}
+	var resp JSApiStreamListResponse
 	if !c.acc.JetStreamEnabled() {
-		s.sendAPIResponse(c, subject, reply, string(msg), JetStreamNotEnabled)
+		resp.Error = jsNotEnabledErr
+		s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(&resp))
 		return
 	}
-	var names []string
 	msets := c.acc.Streams()
 	for _, mset := range msets {
-		names = append(names, mset.Name())
-	}
-	b, err := json.MarshalIndent(names, "", "  ")
-	if err != nil {
-		s.sendAPIResponse(c, subject, reply, string(msg), protoErr(err))
-		return
+		resp.Streams = append(resp.Streams, mset.Name())
 	}
-	s.sendAPIResponse(c, subject, reply, string(msg), string(b))
+	s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(resp))
 }
 
 // Request for information about a stream.
 func (s *Server) jsStreamInfoRequest(sub *subscription, c *client, subject, reply string, msg []byte) {
 	if c == nil || c.acc == nil {
 		return
 	}
+	var resp JSApiStreamInfoResponse
 	if !c.acc.JetStreamEnabled() {
-		s.sendAPIResponse(c, subject, reply, string(msg), JetStreamNotEnabled)
+		resp.Error = jsNotEnabledErr
+		s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(&resp))
 		return
 	}
 	if len(msg) != 0 {
-		s.sendAPIResponse(c, subject, reply, string(msg), JetStreamBadRequest)
+		resp.Error = jsBadRequestErr
+		s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(&resp))
 		return
 	}
 	name := subjectToken(subject, 2)
 	mset, err := c.acc.LookupStream(name)
 	if err != nil {
-		s.sendAPIResponse(c, subject, reply, string(msg), protoErr(err))
-		return
-	}
-	msi := StreamInfo{
-		State:  mset.State(),
-		Config: mset.Config(),
-	}
-	b, err := json.MarshalIndent(msi, "", "  ")
-	if err != nil {
+		resp.Error = jsError(err)
+		s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(&resp))
 		return
 	}
-	s.sendAPIResponse(c, subject, reply, string(msg), string(b))
+	resp.StreamInfo = &StreamInfo{State: mset.State(), Config: mset.Config()}
+	s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(resp))
 }
 
 // Request to delete a stream.
 func (s *Server) jsStreamDeleteRequest(sub *subscription, c *client, subject, reply string, msg []byte) {
 	if c == nil || c.acc == nil {
 		return
 	}
+	var resp JSApiStreamDeleteResponse
 	if !c.acc.JetStreamEnabled() {
-		s.sendAPIResponse(c, subject, reply, string(msg), JetStreamNotEnabled)
+		resp.Error = jsNotEnabledErr
+		s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(&resp))
 		return
 	}
 	if len(msg) != 0 {
```

Comment:
Not relevant to this change, but not sure what is the meaning of this check. Especially that for the delete message request function, we do the opposite check (`if len(msg) == 0`)

Reply:
msg is the arguments that are being sent along as the request. Some API entry points expect no request body, so this marks them as bad requests.

Reply:
Hmm as per the design doc thing I think we should accept empty json as well as nil for those api calls that take no arguments.

Reply:
There is no request to be marshaled for these.

Reply:
So what are you sending?